spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -2238,11 +2239,14 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
 
 		private List<ConsumerRecord<K, V>> createRecordList(final ConsumerRecords<K, V> records) {
 			Iterator<ConsumerRecord<K, V>> iterator = records.iterator();
-			List<ConsumerRecord<K, V>> list = new LinkedList<>();
+			@SuppressWarnings("unchecked") ConsumerRecord<K, V>[] recordsArray =
+					(ConsumerRecord<K, V>[]) Array.newInstance(ConsumerRecord.class, records.count());
+			int index = 0;
 			while (iterator.hasNext()) {
-				list.add(iterator.next());
+				recordsArray[index] = iterator.next();
```

Comment:
Cannot we do like this instead:
```
recordsArray[index++]
```
?

Reply:
Thanks!
That is great discussion and I agree that paradox of choice may cause confusions.

So, I'm accepting your variant of the code to make it more readable.

Please, add your name to the `@author` list of this changed class.